### PR TITLE
Properly run Groovydoc tests, disable some

### PR DESCRIPTION
**Pull request by octylFractal**
Groovydoc 3 & 4 don't check the script flags, so they can't actually be disabled any more.

https://issues.apache.org/jira/browse/GROOVY-10578

### CODE REVIEW

File: subprojects/language-groovy/src/integTest/groovy/org/gradle/groovy/GroovyDocOptionsIntegrationTest.groovy [hunk 1]
```diff
@@ -34,9 +34,16 @@ class GroovyDocOptionsIntegrationTest extends MultiVersionIntegrationSpec {
     private static final String GROOVY_DOC_PUBLIC_PATTERN = /#publicMethod/
 
     private static boolean supportsScriptsInGroovydoc() {
+        def versionNumber = versionNumber
```
Comment (blindpirate) [inline subprojects/language-groovy/src/integTest/groovy/org/gradle/groovy/GroovyDocOptionsIntegrationTest.groovy:37]: Why is this?

**Comment by octylFractal** [comment]
@bot-gradle test this

**Review by blindpirate — APPROVED** [review]
LGTM! Thanks for fixing!

**Comment by octylFractal** [comment]
@bot-gradle test and merge

**Pull request merged**